42 city to coordinate api

vincedbowen opened the pull request:
Finishes the City to Coordinates API
**Functionality:**
This API takes user input from the search page, to connect to the weather API and possibly the flight API in a later PR. The user inputs City and Country names and the call returns the latitude and longitude of said city. This API was needed because we wanted the user to be able to input just City and Country names, but the weather API could only use latitude and longitude. 
**To Review:**
Look over my Axios call, to make sure everything looks appropriate. I tested this with fake cities, and multiple real city, even ones with repeats, to ensure the return is always of high value. To test or implement this yourself head to: [](https://api-ninjas.com/api/geocoding ). Here you can check out documentation and make a free account to test yourself. 
In your .env file copy your API key, found under my account on API-Ninjas and label it:
CITYCOOR_KEY=""

I am aware that my API key is accessible to the public in this commit history. I HAVE SINCE CHANGED MY API KEY AND THAT KEY IS NO LONGER VALID!

Closes #42

StudentKyle commented on All project code/index.js:
```diff
@@ -335,10 +335,37 @@ async function cityToCoordinates(locationInput) {
+        console.log("longitude: ", longitude)
+    }).catch(error => {
+        // Handle errors (API call may have failed!)
+        console.log(error);
```
Can this error be passed to the user in some way using messages? Check how I made the error code -1 in weather API.

StudentKyle replied:
If the page simply doesn't load I guess that's fine for now too, the user can always try again. But this should be addressed at some point.

StudentKyle commented on All project code/index.js:
```diff
@@ -371,7 +398,7 @@ async function searchQuery(locationInput) {
             "precipitation 24 hours"
         ],
-        dataFormat: "json",
+        dataFormat: "html",
```
Good change, this will be helpful for presenting data to the user before we have a complete results page done.

StudentKyle commented on All project code/index.js:
```diff
@@ -335,10 +335,37 @@ async function cityToCoordinates(locationInput) {
     // Perform city/country conversion to latitude/longitude coordinates here.
     const country = locationInput.country;
     const city = locationInput.city;
```
Perhaps some cleaning/parsing of user input would be good, but that can come later as it isn't essential.

StudentKyle commented:
Can you also add "Closes #42" to your PR description so it will close the related issue on merging?

StudentKyle reviewed (changes requested):
Add the closes #42 to the PR description. Other than that, check my feedback in the code, and you can either put those extra features in this PR now or decide to do it in a future PR, as they aren't essential to the functioning feature.

Other than that, everything looks good!

StudentKyle commented on All project code/index.js:
```diff
@@ -335,10 +335,37 @@ async function cityToCoordinates(locationInput) {
+    // axios call to API-Ninja geocoding api to return latitude and longitude from city and country input 
+    await axios({
+        // url for API-Ninja
+        url:'https://api.api-ninjas.com/v1/geocoding?city=',
```
Why does your url have ?city= at the end? I think this should be handled in axios in the params field, unless that's some really specific requirement in the api.

vincedbowen replied:
I had just copied this from the API-Ninja Documentation, but it works without "?city=" so I will remove it.

vincedbowen commented:
I added a bit better error handling, fixed the URL, and added closes #42 to my PR description. As we talked about, this is not great user experience, but it is functional. I will create another issue with better error handling with more informative messages, and an optional state search category to deal with duplicates.

StudentKyle reviewed (approved):
Okay, everything looks good to merge!
